keel-core/src/main/kotlin/com/netflix/spinnaker/keel/_maps.kt

```diff
@@ -0,0 +1,9 @@
+package com.netflix.spinnaker.keel
+
+/**
+ * Filters a map to retain only those entries that have non-null values. Also narrows the
+ * value type on the map.
+ */
+@Suppress("UNCHECKED_CAST")
+fun <K, V : Any> Map<out K, V?>.filterNotNullValues(): Map<K, V> =
+  filterValues { it != null } as Map<K, V>
```

Comment:
Just moved this from a private method in `ImageResolver` as I'm using it somewhere else now too (should be in the Kotlin stdlib if you ask me)